rust/definitions/src/navigation.rs

```diff
@@ -0,0 +1,659 @@
+/// Is used for proper navigation. Variants:
+/// AddSpecs (with associated NetworkSpecsKey), LoadMeta (with associated
+/// NetworkSpecsKey for the first by order network using those metadata),
+/// and LoadTypes
+#[derive(PartialEq, Debug, Clone)]
+pub enum StubNav {
+    AddSpecs { n: NetworkSpecsKey },
+    LoadMeta { l: NetworkSpecsKey },
+    LoadTypes,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum FooterButton {
+    Log,
+    Scan,
+    Keys,
+    Settings,
+    Back,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum RightButton {
+    LogRight,
+    NewSeed,
+    Backup,
+    MultiSelect,
+    NDMenu,
+    TypesInfo,
+    KeyMenu,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ScreenNameType {
+    H1,
+    H4,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ShieldAlert {
+    Past,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum AlertData {
+    Shield { f: Option<ShieldAlert> },
+    ErrorData { f: String },
+    Confirm,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct ActionResult {
+    pub screen_label: String,
+    pub back: bool,
+    pub footer: bool,
+    pub footer_button: Option<FooterButton>,
+    pub right_button: Option<RightButton>,
+    pub screen_name_type: ScreenNameType,
+    pub screen_data: ScreenData,
+    pub modal_data: Option<ModalData>,
+    pub alert_data: Option<AlertData>,
+}
+
+#[derive(Clone, PartialEq)]
+pub struct LogScreenEntry {
+    pub timestamp: String,
+    pub events: Vec<Event>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ScreenData {
+    Scan,
+    Keys { f: MKeys },
+    Settings { f: MSettings },
+    Log { f: MLog },
+    LogDetails { f: MLogDetails },
+    Transaction { f: MTransaction },
+    SeedSelector { f: MSeeds },
+    KeyDetails { f: MKeyDetails },
+    NewSeed { f: MNewSeed },
+    RecoverSeedName { f: MRecoverSeedName },
+    RecoverSeedPhrase { f: MRecoverSeedPhrase },
+    DeriveKey { f: MDeriveKey },
+    VVerifier { f: MVerifierDetails },
+    ManageNetworks { f: MManageNetworks },
+    NNetworkDetails { f: MNetworkDetails },
+    SignSufficientCrypto { f: MSignSufficientCrypto },
+    SelectSeedForBackup { f: MSeeds },
+    Documents,
+    KeyDetailsMulti { f: MKeyDetailsMulti },
+}
+
+#[derive(Clone, PartialEq)]
+pub struct Identity {
+    pub seed_name: String,
+    pub address_key: String,
+    pub public_key: String,
+    pub identicon: Vec<u8>,
+    pub has_pwd: bool,
+    pub path: String,
+    pub is_multiselect: bool,
+    pub base58: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MKeysCard {
+    pub address_key: String,
+    pub base58: String,
+    pub identicon: Vec<u8>,
+    pub has_pwd: bool,
+    pub path: String,
+    pub swiped: bool,
+    pub multiselect: bool,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MNetworkCard {
+    pub title: String,
+    pub logo: String,
+}
+
+// TODO: This has to have a custom default.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MSeedKeyCard {
```

Comment:
`Address` should not have `multiselect` key. `MSeedKeyCard` should probably have fields `address` and `mode`.
enum Mode {
    Plain,
    Swiped(_),
    Multiselect(_),
}
`swiped` and `multiselect` exclude each other and if we go for proper types, those bools that were for json simplicity only are no longer needed. This can wait for later, as now the whole thing works properly.